### PR TITLE
[dif/pwrmgr] add alert test functionality

### LINKED ISSUE #7641: [dif/pwrmgr] missing alert test functionality
There is an alert test register that forces an alert signal to be activated for testing. The DIF currently does not support exercising this functionality, however, the DIFs for other IP blocks (e.g., the AES block) do.

To fix, an additional function must be added to the DIF.

### PR DESCRIPTION
**Pull request by timothytrippel**
This fixes #7641.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
@alphan please review

**Review by moidx — APPROVED** [review]

**Review by sriyerg — APPROVED** [review]

File: sw/device/lib/dif/dif_pwrmgr_unittest.cc [hunk 1]
```diff
@@ -894,5 +894,23 @@ TEST_F(IrqTest, RestoreAll) {
   EXPECT_EQ(dif_pwrmgr_irq_restore_all(&pwrmgr_, snapshot), kDifPwrmgrOk);
 }
 
+class AlertTest : public DifPwrmgrInitialized {};
+
+TEST_F(AlertTest, ForceBadArgs) {
+  dif_pwrmgr_alert_t alert = kDifPwrmgrAlertFatalFault;
+
+  EXPECT_EQ(dif_pwrmgr_alert_force(nullptr, alert), kDifPwrmgrBadArg);
+  EXPECT_EQ(dif_pwrmgr_alert_force(&pwrmgr_, (dif_pwrmgr_alert_t)1),
```
Comment (alphan) [inline sw/device/lib/dif/dif_pwrmgr_unittest.cc:903]: nit:
```suggestion
  EXPECT_EQ(dif_pwrmgr_alert_force(&pwrmgr_, static_cast<dif_pwrmgr_alert_t>(1)),
```

File: sw/device/lib/dif/dif_pwrmgr_unittest.cc [hunk 2]
```diff
@@ -894,5 +894,23 @@ TEST_F(IrqTest, RestoreAll) {
   EXPECT_EQ(dif_pwrmgr_irq_restore_all(&pwrmgr_, snapshot), kDifPwrmgrOk);
 }
 
+class AlertTest : public DifPwrmgrInitialized {};
+
+TEST_F(AlertTest, ForceBadArgs) {
+  dif_pwrmgr_alert_t alert = kDifPwrmgrAlertFatalFault;
```
Comment (alphan) [inline sw/device/lib/dif/dif_pwrmgr_unittest.cc:900]: nit: can be inlined, or renamed to `valid_alert` to indicate purpose.

**Review by alphan — APPROVED** [review]
Thanks @timothytrippel!

File: sw/device/lib/dif/dif_pwrmgr_unittest.cc (hunk 1 above)
Comment (timothytrippel) [reply sw/device/lib/dif/dif_pwrmgr_unittest.cc]: Fixed.

File: sw/device/lib/dif/dif_pwrmgr_unittest.cc (hunk 2 above)
Comment (timothytrippel) [reply sw/device/lib/dif/dif_pwrmgr_unittest.cc]: Fixed.

**Pull request merged**